File: PrimeGenerating/SeiveOfErasthosthenes/SOE.java
import java.lang.Math;
public class SOE {
   public static void main(String args[ ])
   {
    long start_milli=System.currentTimeMillis();
    long start_nano=System.nanoTime();
    long end_milli;
    long end_nano;

	int range=10000;
	boolean array[]=new boolean[range];
	int square_of_num=(int)Math.sqrt(range);
	for(int start=2;start<= square_of_num;start++)
	{
	   if(!array[start])
	   {
    for(int condition = start * start ; condition < range ; condition +=start )
		    array[condition]=true;
	   }
    }
	int count=0;
       	for(int index=2;index<array.length;index++)
         if(!array[index])
		 {
			 
			 count++;
		 }
	System.out.println(count);

    end_milli=System.currentTimeMillis();
    end_nano=System.nanoTime();
    System.out.println("Time : "+(end_milli-start_milli)/1000.0);
    System.out.println((end_nano-start_nano));
	}
}

File: PrimeGenerating/SeiveOfErasthosthenes/soe.cpp
#include <iostream>
#include <math.h>
using namespace std;
int main() 
{
    int a=1000000;
    bool arr[a+1]={true};
    long long int b=sqrt(a);
    for(long long int i=2;i<=b;i++)
    {
        if(arr[i]==0)
        {
            for(long long int j=i*i;j<a;j+=i)
                arr[j]=1;
        }
    }
    int count=0;
    for(long long int i=2;i<a;i++)
        if(arr[i]==0)
            count+=1;
    cout<<count<<endl;
    return 0;
}

